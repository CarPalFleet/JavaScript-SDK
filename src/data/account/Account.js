/**
 * @fileoverview This file contains all account related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import { apiResponseErrorHandler } from '../utility/Util';

/**
 * Request reset password token
 * @param {string} email
 * @return {object} Promise resolve/reject
 */
export const resetPasswordRequestAsync = async (email) => {
  try {
    const response = await axios({
      method: 'POST',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        email,
      },
    });
    return camelize(response);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Reset password with valid token
 * @param {string} token
 * @param {string} email
 * @param {string} password
 * @param {string} passwordConfirmation
 * @return {object} Promise resolve/reject
 */
export const resetPasswordAsync = async (
  token,
  email,
  password,
  passwordConfirmation
) => {
  try {
    const response = await axios({
      method: 'PUT',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
        email,
        password,
        passwordConfirmation,
      },
    });
    return camelize(response.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Validate Reset Password Token
 * @param {string} token
 * @return {object} Promise resolve/reject
 */
export const validateResetPasswordTokenAsync = async (token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: endpoints.API_V3.VALIDATE_PASSWORD_RESET_TOKEN.replace(
        '{token}',
        token
      ),
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
      },
    });
    return camelize(response.data.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};
