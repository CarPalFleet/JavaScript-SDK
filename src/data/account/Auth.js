import axios from 'axios';
import endpoints, { cid, secret } from '../Endpoint';

export const getTokenAsync = async (email, password)=>{
    try{
        const response = await axios({method: 'post', 
                                      url: endpoints.OAUTH,
                                      headers: {'Content-Type': 'application/json'},
                                      data: {
                                        username: email,
                                        password,
                                        grant_type: "password",
                                        client_id: cid,
                                        client_secret: secret,
                                        scope: ""
                                      }})
        return response.data.data;     
    }catch(e){
        return null
    }
}            