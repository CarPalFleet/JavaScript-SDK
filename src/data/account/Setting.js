/**
 * @fileoverview This file contains all Settings related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import {apiResponseErrorHandler} from '../utility/Util';

/** Retrieve User's settings
 * There are 3 setting types in the setting table
 * 1. routing, 2. my-order, 3. driver-list
 * In routing type, it includes user time line setting (15 min, 30 min, 45 min etc.)
 * Retrieve table settings from my-order type OR driver-list.
 * @param {integer} userId
 * @param {string} type # routing, my-order, driver-list
 * @param {string} token
 * @return {object} promise (resolve/reject)
 */
export const getUserSettingsAsync = async (userId, type, token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: `${endpoints.USER_SETTINGS.replace(
        '{0}',
        userId
      )}?type=${type}`,
      headers: {Authorization: token},
    });
    return camelize(response.data);
  } catch (e) {

    return apiResponseErrorHandler(e);
  }
};
