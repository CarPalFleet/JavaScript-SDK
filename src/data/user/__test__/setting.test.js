import { getTokenAsync } from '../../account/Auth';
import {
  getUserSettingMyOrdersColumnAsync,
  putUserSettingMyOrdersColumn,
  getUserSettingsAsync,
} from '../Setting';
import CONFIG from './Config';

describe('Test getUserSettingsAsync function', async () => {
  let token;

  beforeAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    token = await getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
  });

  it('should return user settings status 200', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    try {
      const response = await getUserSettingsAsync(token.accessToken);
      expect(response).toMatchSnapshot();
    } catch (error) {
      expect(error).toHaveProperty('statusCode');
    }
  });
});

describe('Test getUserSettingMyOrdersColumnAsync function', async () => {
  let token;

  beforeAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    token = await getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
  });

  it('should get user settings', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    try {
      const response = await getUserSettingMyOrdersColumnAsync(
        token.accessToken
      );
      expect(response.data).toMatchSnapshot();
    } catch (error) {
      expect(error).toHaveProperty('statusCode');
    }
  });
});

describe('test updating user settings', async () => {
  let token;
  beforeAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    token = await getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
  });

  it('should return 200', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    const response = await putUserSettingMyOrdersColumn(
      [
        {
          key: 'customerOrderNumber',
          isFixed: false,
          isVisible: true,
        },
      ],
      token.accessToken
    );

    expect(response).toHaveProperty('status', 200);
    expect(response.data).toHaveProperty('myOrdersColumns', [
      { isFixed: false, isVisible: true, key: 'customerOrderNumber' },
    ]);
    expect(response.data.myOrdersColumns[0]).toHaveProperty('isVisible', true);
  });
});
