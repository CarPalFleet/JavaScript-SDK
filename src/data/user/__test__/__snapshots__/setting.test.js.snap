// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getUserSettingMyOrdersColumnAsync function should get user settings 1`] = `
Object {
  "myOrdersColumns": Array [
    Object {
      "isFixed": false,
      "isVisible": true,
      "key": "customerOrderNumber",
    },
  ],
}
`;

exports[`Test getUserSettingsAsync function should return user settings status 200 1`] = `
Array [
  Object {
    "defaultLanguageCode": "en",
  },
  Object {
    "defaultLanguageCode": "en",
  },
  Object {
    "myOrdersColumns": Array [
      Object {
        "isFixed": false,
        "isVisible": true,
        "key": "customerOrderNumber",
      },
    ],
  },
]
`;
